ui-participant/src/api/api.tsx

```diff
@@ -150,12 +151,16 @@ export default {
     }
   },
 
-  async processJsonResponse(response: Response) {
+  /** get the json from the response and alert and log any errors */
+  async processJsonResponse(response: Response, opts: {alertErrors: boolean} = { alertErrors: true }) {
     const obj = await response.json()
     if (response.ok) {
       return obj
     }
-    return Promise.reject(response)
+    if (opts.alertErrors) {
+      defaultApiErrorHandle(obj)
+    }
+    return Promise.reject(obj)
   },
```

Comment:
When adding additional options, all will have to be specified, but you may want to only specify one and let the others keep their defaults. Something like this will allow for fully optional opts with the correct defaults:
```suggestion
  async processJsonResponse(response: Response, opts?: { alertErrors?: boolean }) {
    const defaultOpts = { alertErrors: true }
    const { alertErrors } = { ...defaultOpts, ...opts }
    
    const obj = await response.json()
    if (response.ok) {
      return obj
    }
    if (alertErrors) {
      defaultApiErrorHandle(obj)
    }
    return Promise.reject(obj)
  },
```

Reply:
cool, will use that once this inevitably grows :)